tests/util_tests.py

```diff
@@ -29,3 +29,10 @@ def test_is_timestamp(self):
         overflow_timestamp_int = int(overflow_timestamp_float)
         with self.assertRaises((OverflowError, ValueError)):
             util.is_timestamp(overflow_timestamp_int)
+
+    def test_iso_gregorian(self):
```

Comment:
We should probably test more edge cases such as those listed here: https://stackoverflow.com/q/304256/.